react/components/utils/useDeviceHook.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+import { useMedia } from 'use-media'
+
+export enum Device {
+  phone = 'phone',
+  tablet = 'tablet',
+  desktop = 'desktop',
+}
+
+const withDevice = Component => {
+  function WithDevice(props) {
+    // Taken at https://github.com/vtex-apps/device-detector/blob/master/react/useDevice.tsx#L22
```

Comment:
This is not a `hook`, it is a `hoc`. it would be better to rename the file to `withDeviceHoc`.